rules/TypeDeclaration/AlreadyAssignDetector/ConstructorAssignDetector.php

```diff
@@ -106,6 +109,70 @@ private function decorateFirstLevelStatementAttribute(array $classMethods): void
                 if ($methodStmt instanceof Expression) {
                     $methodStmt->expr->setAttribute(self::IS_FIRST_LEVEL_STATEMENT, true);
                 }
+
+                // support top level if statements
+                if ($methodStmt instanceof Node\Stmt\If_) {
+                    $alwaysAssignedProperties = [];
+                    foreach($methodStmt->stmts as $ifStmt) {
+                        $assign = null;
+                        if ($ifStmt instanceof Expression && $ifStmt->expr instanceof Assign) {
+                            $assign = $ifStmt->expr;
+                        }
+
+                        if ($assign instanceof Assign && $assign->var instanceof Expr\PropertyFetch) {
+                            $name = $this->nodeNameResolver->getName($assign->var);
+                            $alwaysAssignedProperties[$name] = $assign;
+                        }
+                    }
+
+                    /*
```

Comment:
the PR as is works for IF/ELSE constructions, but not yet for IF/ELSEIF/ELSE constructions.

I can do this last piece after we agreed on the approach.

any feedback? should I proceed?

Reply:
I am curious if there is already PHPStan attribute for "always assigned, like `assigned_to`, ref https://github.com/search?q=repo%3Arectorphp%2Frector-src%20ASSIGNED_TO&type=code